packages/vendure-plugin-customer-managed-groups/src/customer-managed-groups.service.ts

```diff
@@ -206,7 +206,7 @@ export class CustomerManagedGroupsService {
     ctx: RequestContext,
     userId: ID
   ): Promise<CustomerWithCustomFields> {
-    const customerRepo = getRepository(Customer);
+    const customerRepo = this.dataSource.getRepository(Customer);
```

Comment:
Hi @dalyathan, we can use Vendure's transactional connection here, when you pass it the CTX as well its not deprecated:

```ts
constructor(
  private connection: TransactionalConnection
  ){}
  
const repo = await  this.connection.getRepository(ctx, Customer);
```

This also handles transaction logic for us.